kombu_redis_priority/scheduling/prioritized_levels.py

```diff
@@ -0,0 +1,121 @@
+"""
+Prioritized Levels scheduler for the queues.
+
+Given a configuration of queues encoded as:
+{
+    LEVEL: [QUEUE]
+}
+where LEVEL is a numeric value indicating priority (smaller
+first) and [QUEUE] indicates a list of queue names, the
+scheduler will walk from smallest level to highest level,
+only advancing levels if the smaller levels are empty.
+Within levels, the queues are rotated in round robin
+fashion.
+
+To honor the prefernece for the lower levels, we will move
+back to the lowest level when we do a full rotation of the
+current level.
+
+This is done to support the asynchronous nature in which
+kombu pulls tasks.
+"""
+import sys
+from collections import defaultdict
+from kombu.utils.scheduling import cycle_by_name
+from .base import QueueScheduler
+
+HIGHEST_LEVEL = sys.maxsize
```

Comment:
I don't know if we care at all about this edge case, but since Python will either invisibly convert numbers into `long` types (python2) or just treat really big numbers as regular `int`s (python3), this doesn't make any actual guarantees about being the last element popped off the queues.

But it's kind of a silly edge case where you're using the interface very strangely.